requirements-client.txt

```
@@ -15,7 +15,8 @@ jsonschema >= 3.2.0, < 5.0.0
 orjson >= 3.7, < 4.0
 packaging >= 21.3, < 24.3
 pathspec >= 0.8.0
-pendulum >= 2.1.2, < 3.0.0
+pendulum < 3.0; python_version < '3.12'
+pendulum >= 3.0.0; python_version >= '3.12'
```

```suggestion
pendulum >= 3.0.0, <4; python_version >= '3.12'
```

Just in case pendulum picks up its release cadence :)